File: src/backend/app/models/skill.py
from typing import Literal
from pydantic import BaseModel

SkillLevel = Literal["beginner", "intermediate", "advanced", "expert"]


class Skill(BaseModel):
    id: str
    name: str
    level: SkillLevel
    category: str | None = None


class SkillGap(BaseModel):
    skill: Skill
    targetLevel: SkillLevel
    priority: Literal["low", "medium", "high"]

File: src/frontend/types/skill.ts
export interface Skill {
  id: string;
  name: string;
  level: "beginner" | "intermediate" | "advanced" | "expert";
  category?: string;
}

export interface SkillGap {
  skill: Skill;
  targetLevel: Skill["level"];
  priority: "low" | "medium" | "high";
}
